src/track/tracktiminginfo.cpp

```diff
@@ -0,0 +1,82 @@
+#include "track/tracktiminginfo.h"
+
+TrackTimingInfo::TrackTimingInfo(TrackPointer pTrack) :
+        m_pElapsedTimer(new TrackTimers::ElapsedTimerQt()),
+        m_pTimer(new TrackTimers::GUITickTimer()),
+        m_pTrackPtr(pTrack), 
+        m_playedMs(0),
+        m_isTrackScrobbable(false)
+{
+    connect(m_pTimer.get(),SIGNAL(timeout()),
+            this,SLOT(slotCheckIfScrobbable()));
+}
+
+void TrackTimingInfo::pausePlayedTime() {    
+    if (m_pElapsedTimer->isValid()) {
+        m_playedMs += m_pElapsedTimer->elapsed();
+        m_pElapsedTimer->invalidate();
+        QObject::disconnect(m_pTimer.get(),SIGNAL(timeout()),
```

Comment:
Disconnect locks a muted internally. You should not connect and disconnect signals after initialize.
Do you use Qt5? I am suffering from bufferunderflows. Maybe this is related.
Does it still deadlock if you remove this line? 
Is there a way around using a timer signal here?

Reply:
Unfortunately, there's a deadlock even if I remove every single connect and disconnect call inside that class. I tried using valgrind but it's giving me an illegal instruction error. I'm going to investigate further. Also, I don't use Qt5, I will check on that.

Reply:
Qt5 will introduce new problems. I would stick with Qt4 in this case. 
I will have a look into this soon today.